source/grid/tria_description.cc

```diff
@@ -525,6 +526,382 @@ namespace TriangulationDescription
+          });
+          temp.erase(std::unique(temp.begin(),
+                                 temp.end(),
+                                 [](const auto &a, const auto &b) {
+                                   return std::get<0>(a) == std::get<0>(b);
+                                 }),
+                     temp.end());
+          std::sort(temp.begin(), temp.end(), [](const auto &a, const auto &b) {
+            return std::get<2>(a) < std::get<2>(b);
+          });
+
+          this->coarse_cell_index_to_coarse_cell_id.resize(temp.size());
+          this->coarse_cells.resize(temp.size());
+
+          for (unsigned int i = 0; i < temp.size(); ++i)
+            {
+              this->coarse_cell_index_to_coarse_cell_id[i] =
+                std::get<0>(temp[i]);
+              this->coarse_cells[i] = std::get<1>(temp[i]);
+            }
+        }
+
+        // make coarse cell vertices unique
+        {
+          std::sort(this->coarse_cell_vertices.begin(),
+                    this->coarse_cell_vertices.end(),
+                    [](const auto &a, const auto &b) {
+                      return a.first < b.first;
+                    });
+          this->coarse_cell_vertices.erase(
+            std::unique(this->coarse_cell_vertices.begin(),
+                        this->coarse_cell_vertices.end(),
+                        [](const auto &a, const auto &b) {
+                          return a.first == b.first;
+                        }),
+            this->coarse_cell_vertices.end());
+        }
+
+        // make cells unique
+        for (unsigned int i = 0; i < this->cell_infos.size(); ++i)
+          {
+            std::sort(this->cell_infos[i].begin(),
+                      this->cell_infos[i].end(),
+                      [](const auto &a, const auto &b) { return a.id < b.id; });
+            this->cell_infos[i].erase(std::unique(this->cell_infos[i].begin(),
+                                                  this->cell_infos[i].end(),
+                                                  [](const auto &a,
+                                                     const auto &b) {
+                                                    return a.id == b.id;
+                                                  }),
+                                      this->cell_infos[i].end());
+          }
+      }
+
+      Description<dim, spacedim>
+      convert(const MPI_Comm comm,
+              const typename Triangulation<dim, spacedim>::MeshSmoothing
+                mesh_smoothing)
+      {
+        Description<dim, spacedim> description;
+
+        // copy communicator
+        description.comm = comm;
+
+        // set settings (no multigrid levels are set for now)
+        description.settings =
+          TriangulationDescription::Settings::default_setting;
+
+        // use mesh smoothing from base triangulation
+        description.smoothing = mesh_smoothing;
+
+        std::map<unsigned int, unsigned int> map;
```

Comment:
Aren't you using the `map` below as a `std::vector<unsigned int> map(coarse_cell_vertices.size());`? Or which index range does `coarse_cell_vertices[i].first` span?

Reply:
The size is the number of vertices of the coarse grid of p:d:T. The reason why I switched here to a map is that I assume that we only have small number of locally relevant vertices here (only that we need to construct the local part of p:f:T). Do you still insist only switching the map to a vector?

Reply:
Ah, I see, I forgot the selection of the relevant vertices and skipping "artificial" ones. It remain that way then.